src/site/content/en/learn/css/layout/index.md

```diff
@@ -0,0 +1,401 @@
+For example,
+setting the `display` property to `display: flex` makes the box a block-level box,
+and also converts its children to flex items.
+This enables the flex properties that control alignment, ordering and flow.
+
+## Flexbox and Grid
+
+There are two main layout mechanisms that create layout rules for multiple elements, *flexbox* and *grid*.
+They share similarities, but are designed to solve different layout problems.
+
+@@TODO: link to flexbox and grid lessons above
+
+{% Aside %}
+We will be going into much more detail for both of these in future lessons, but here is a high-level overview of what both are and what they are useful for.
+{% endAside %}
+
+### Flexbox
+
+```css
+.my-element {
+	display: flex;
+}
+```
+
+Flexbox is a layout mechanism for one-dimensional layouts.
+Layout across a single axis, either horizontally or vertically.
+By default, flexbox will align the element's children next to each other,
+in the inline direction,
+and stretch them in the block direction, so they're all the same height.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'rNjxmor'
+} %}
+
+
+Items will stay on the same axis and not wrap when they run out of space.
+Instead they will try to squash onto the same line as each other.
+This behaviour can be changed using the `align-items`, `justify-content` and `flex-wrap` properties.
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'jOyWLmg'
+} %}
+
+
+Flexbox also converts the child elements to be **flex items**,
+which means you can write rules on how they behave inside a flex container.
+You can change alignment, order and justification on an individual item.
+You can also change how it shrinks or grows using  the `flex` property.
+
+```css
+.my-element div {
+ 	flex: 1 0 auto;
+}
+```
+
+The `flex` property is a shorthand for `flex-grow`, `flex-shrink` and `flex-basis`.
+ You can expand the above example like this:
+
+```css
+.my-element div {
+ flex-grow: 1;
+ flex-shrink: 0;
+ flex-basis: auto;
+}
+```
+
+Developers provide these low-level rules to hint a browser how the layout should behave
+when it is challenged by content and viewport dimensions.
+This makes it a very useful mechanism for responsive web design.
+
+### Grid
+
+```css
+.my-element {
+	display: grid;
+}
+```
+
+Grid is similar in a lot of ways to **flexbox**,
+but it is designed to control multi-axis layouts instead of single-axis layouts (vertical or horizontal space).
+
+Grid enables you to write layout rules on an element that has `display: grid`,
+and introduces a few new primitives for layout styling,
+such as the `repeat()` and `minmax()` functions.
+One useful grid unit is the  `fr` unit—which is a fraction of remaining space—you can build traditional 12 column grids,
+with a gap between each item, with 3 CSS properties:
+
+```css
+.my-element {
+  display: grid;
+  grid-template-columns: repeat(12, 1fr);
+  gap: 1rem;
+}
+```
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'rNjxGVz'
```

Comment:
This demo and the 2 after look a little funky in the demo view from the module. Maybe we can reduce the number of items so it fits the space a bit better?

Reply:
Good catches mate :)